### PR TITLE
fix : Show download dialog on emulator

### LINKED ISSUE #1868: App not showing download dialog on emulator
**Actual Behaviour**

Currently we are using `ping` utility to check active internet connection. But ping don't work on Android emulator. Because of it app is not showing download dialogue on emualtor.

https://stackoverflow.com/questions/14234873/can-i-use-ping-command-in-the-android-emulator
 
**Expected Behaviour**

User other method for checking active internet connection. i.e. Using HttpUrlConnection and checking response code. 

**Would you like to work on the issue?**

Yes

**Target Milestone to solve the issue**

[Milestone 14](https://github.com/fossasia/open-event-android/milestone/37)

### PR DESCRIPTION
**Pull request by Shailesh351**
Fixes #1868 

Changes:

- Remove active internet connection check
- Remove ShowNotificationSnackbar class which is **redundant** because snackbar is shown using event bus(DownloadEvent).
- Fixed error in getting facebook page id
- Created View util method for SwipeRefreshLayout to setRefreshing state.
- Update sample api-link

### CODE REVIEW

File: android/app/src/main/assets/config.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "email": "dev@fossasia.org",
   "app-name": "Open Event",
-  "api-link": "https://open-event-api.herokuapp.com/v1/events/59/",
+  "api-link": "https://api.eventyay.com/v1/events/20",
```
Comment (iamareebjamal) [inline android/app/src/main/assets/config.json]: It's better to depend on develop server instead

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

**Comment by Shailesh351** [comment]
@iamareebjamal @mejariamol Please review

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java [hunk 2]
```diff
@@ -406,15 +406,16 @@ private void downloadPageId() {
                     .subscribe(facebookPageId -> {
                         String id = facebookPageId.getId();
                         SharedPreferencesUtil.putString(ConstantStrings.FACEBOOK_PAGE_ID, id);
-                    });
+                    }, throwable -> Timber.d("Facebook page id download failed: " + throwable.toString()));
```
Comment (mejariamol) [inline android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java]: please check if the indentation is correct.

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java [hunk 3]
```diff
@@ -639,6 +622,11 @@ public void showErrorDialog(String errorType, String errorDesc) {
         builder.show();
     }
 
+    public void showErrorSnackbar(String errorType, String errorDesc) {
```
Comment (mejariamol) [inline android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java:627]: may just pass the error as a parameter.

File: android/app/src/main/java/org/fossasia/openevent/fragments/LocationsFragment.java [hunk 4]
```diff
@@ -178,18 +182,35 @@ public void onDestroyView() {
 
     @Subscribe
     public void onLocationsDownloadDone(MicrolocationDownloadEvent event) {
-        if(swipeRefreshLayout == null)
-            return;
+        if (swipeRefreshLayout != null && swipeRefreshLayout.isRefreshing())
```
Comment (mejariamol) [inline android/app/src/main/java/org/fossasia/openevent/fragments/LocationsFragment.java]: this part seems repeating at more than one places. Have a utility method which takes in the refresh layout and does the action.

File: android/app/src/main/java/org/fossasia/openevent/utils/Notification.java [hunk 5]
```diff
@@ -9,65 +9,31 @@
...
-
-
-    public void buildNotification(){
-        if(context == null)
+    public void buildNotification() {
+        if (context == null)
             return;
 
         ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
         android.net.NetworkInfo isWifi = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
```
Comment (mejariamol) [inline android/app/src/main/java/org/fossasia/openevent/utils/Notification.java]: change the field name `isWifi` to something else, it seems a boolean by name. may be just `Wifi` is enough.

**Review by mejariamol — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java (hunk 3 above)
Comment (Shailesh351) [reply android/app/src/main/java/org/fossasia/openevent/activities/MainActivity.java:627]: Snackbar will not able to show whole error message in one line. That's why i'm using it.

**Comment by Shailesh351** [comment]
@mejariamol @iamareebjamal Updated please review.

**Review by iamareebjamal — DISMISSED** [review]

**Review by mejariamol — APPROVED** [review]

**Comment by mejariamol** [comment]
@Shailesh351 Please squeeze your commits into one. It is a good practice to have one commit per PR.

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**